server/src/main/java/org/elasticsearch/readiness/ReadinessService.java

```diff
@@ -194,6 +194,12 @@ protected void doStop() {
     // package private for testing
     synchronized void stopListener() {
         assert enabled(environment);
+
+        // don't spam the logs with repeated false, if we were never active
+        if (ready() == false) {
+            return;
+        }
```

Comment:
I'm uncomfortable having this in the same commit as a test fix, at least not without much deeper analysis. If it's just about logging, I think it'd be best to revert it from this PR and do this separately so it doesn't get backported.

Reply:
Sounds good, I've reverted it and I'll follow-up with a new PR. Thanks!